### PR TITLE
Fix Marionette Issues

### PR DESCRIPTION
**Pull request by samccone**
started fixing a bug #1290 .. came across many fixes so I did them

### CODE REVIEW

**Comment by samccone** [comment]
:wave: CI :)

**Comment by samccone** [comment]
I :sparkling_heart: the new test suite

File: examples/backbone_marionette/js/TodoMVC.TodoList.js [hunk 1]
```diff
@@ -17,19 +17,24 @@ TodoMVC.module('TodoList', function (TodoList, App, Backbone, Marionette) {
...
 		start: function () {
 			this.showHeader(this.todoList);
 			this.showFooter(this.todoList);
 			this.showTodoList(this.todoList);
+			this.todoList.on('all', this.updateHiddenElements, this);
 			this.todoList.fetch();
 		},
 
+		updateHiddenElements: function () {
+			$('#main, #footer')[this.todoList.length == 0 ? 'hide' : 'show']();
```
Comment (arthurvr) [inline examples/backbone_marionette/js/TodoMVC.TodoList.js]: What's wrong with
```js
$('#main, #footer').toggle(!!this.todoList.length);
```

**Comment by arthurvr** [comment]
Updates to the test suite shouldn't go in a PR to fix marionette things.

> I :sparkling_heart: the new test suite

Me too! :heart_eyes:

File: examples/backbone_marionette/js/TodoMVC.TodoList.js (hunk 1 above)
Comment (samccone) [inline examples/backbone_marionette/js/TodoMVC.TodoList.js]: totally
http://api.jquery.com/toggle/#toggle-display

wow did not know this took a boolean :+1:

**Comment by samccone** [comment]
kk all updated @arthurvr

**Pull request merged**